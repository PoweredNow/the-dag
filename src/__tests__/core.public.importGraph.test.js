const addEdges = require('../core/public/addEdges');
const addNodes = require('../core/public/addNodes');
const importGraph = require('../core/public/importGraph');

const getStateManipulators = require('../core/private/_getStateManipulators');
const { getState } = require('../core/private/_state');

describe('core.public.importGraph', () => {
  beforeAll(() => {
    getStateManipulators().setState(graph => {
      graph.nodes = {};
      graph.edges = {};
      return graph;
    });
  });
  test('exports', () => {
    expect(importGraph).toMatchSnapshot();
  });
  test('importGraph works with right arguments', () => {
    const stateHandlers = getStateManipulators();
    const inputGraphWithDifferentFormat = {
      nodes: [1, 2, 3, 4, 5],
      edges: [
        { source: 1, target: 2 },
        { source: 2, target: 3 },
        { source: 2, target: 4 }
      ]
    };
    const graphReducers = {
      nodeIDGenerator: node => node,
      edgeSourceIDGenerator: edge => edge.source,
      edgeTargetIDGenerator: edge => edge.target
    };
    importGraph(
      Object.assign({}, inputGraphWithDifferentFormat, graphReducers)
    );
    expect(stateHandlers.getState()).toMatchSnapshot();
  });

  test('importGraph works without reducers in right format', () => {
    const stateHandlers = getStateManipulators();
    const inputGraphWithDifferentFormat = {
      nodes: [
        { nodeID: 1 },
        { nodeID: 2 },
        { nodeID: 3 },
        { nodeID: 4 },
        { nodeID: 5 }
      ],
      edges: [
        { source: 1, target: 2 },
        { source: 2, target: 3 },
        { source: 2, target: 4 }
      ]
    };
    const graphReducers = {};
    importGraph(
      Object.assign({}, inputGraphWithDifferentFormat, graphReducers)
    );
    expect(stateHandlers.getState()).toMatchSnapshot();
  });
});
