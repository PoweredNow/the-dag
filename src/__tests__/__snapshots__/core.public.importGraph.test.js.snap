// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core.public.importGraph exports 1`] = `[Function]`;

exports[`core.public.importGraph importGraph works with right arguments 1`] = `
Object {
  "edges": Object {
    "1_2": Object {
      "edgeData": Object {},
      "edgeID": "1_2",
      "sourceID": 1,
      "targetID": 2,
    },
    "2_3": Object {
      "edgeData": Object {},
      "edgeID": "2_3",
      "sourceID": 2,
      "targetID": 3,
    },
    "2_4": Object {
      "edgeData": Object {},
      "edgeID": "2_4",
      "sourceID": 2,
      "targetID": 4,
    },
  },
  "nodes": Object {
    "1": Object {
      "nodeData": 1,
      "nodeID": 1,
      "possibleTargets": Array [
        2,
      ],
    },
    "2": Object {
      "nodeData": 2,
      "nodeID": 2,
      "possibleSources": Array [
        1,
      ],
      "possibleTargets": Array [
        3,
        4,
      ],
    },
    "3": Object {
      "nodeData": 3,
      "nodeID": 3,
      "possibleSources": Array [
        2,
      ],
      "possibleTargets": Array [],
    },
    "4": Object {
      "nodeData": 4,
      "nodeID": 4,
      "possibleSources": Array [
        2,
      ],
      "possibleTargets": Array [],
    },
    "5": Object {
      "nodeData": 5,
      "nodeID": 5,
      "possibleTargets": Array [],
    },
  },
}
`;

exports[`core.public.importGraph importGraph works without reducers in right format 1`] = `
Object {
  "edges": Object {
    "1_2": Object {
      "edgeData": Object {},
      "edgeID": "1_2",
      "sourceID": 1,
      "targetID": 2,
    },
    "2_3": Object {
      "edgeData": Object {},
      "edgeID": "2_3",
      "sourceID": 2,
      "targetID": 3,
    },
    "2_4": Object {
      "edgeData": Object {},
      "edgeID": "2_4",
      "sourceID": 2,
      "targetID": 4,
    },
  },
  "nodes": Object {
    "1": Object {
      "nodeData": 1,
      "nodeID": 1,
      "possibleTargets": Array [
        2,
      ],
    },
    "2": Object {
      "nodeData": 2,
      "nodeID": 2,
      "possibleSources": Array [
        1,
      ],
      "possibleTargets": Array [
        3,
        4,
      ],
    },
    "3": Object {
      "nodeData": 3,
      "nodeID": 3,
      "possibleSources": Array [
        2,
      ],
      "possibleTargets": Array [],
    },
    "4": Object {
      "nodeData": 4,
      "nodeID": 4,
      "possibleSources": Array [
        2,
      ],
      "possibleTargets": Array [],
    },
    "5": Object {
      "nodeData": 5,
      "nodeID": 5,
      "possibleTargets": Array [],
    },
    "[object Object]": Object {
      "nodeData": Object {
        "nodeID": 1,
      },
      "nodeID": Object {
        "nodeID": 1,
      },
      "possibleTargets": Array [],
    },
  },
}
`;
